pyproject.toml

```diff
@@ -1,104 +1,176 @@
 [build-system]
 build-backend = "hatchling.build"
-requires = ["hatchling", "hatch-vcs"]
+requires = [ "hatch-vcs", "hatchling" ]
 
 [project]
 name = "mudata"
 description = "Multimodal data"
-requires-python = ">= 3.10"
-license = "BSD-3-Clause"
-authors = [
-    { name = "Danila Bredikhin" },
-]
+readme = "README.md"
+license = { file = "LICENSE" }
 maintainers = [
-    { name = "Danila Bredikhin", email = "danila@stanford.edu" },
+  { name = "Danila Bredikhin", email = "danila@stanford.edu" },
 ]
-readme = "README.md"
-classifiers = [
-    "Programming Language :: Python :: 3",
-    "License :: OSI Approved :: BSD License",
-    "Operating System :: OS Independent",
-    "Development Status :: 3 - Alpha",
-    "Topic :: Scientific/Engineering :: Bio-Informatics",
-    "Intended Audience :: Science/Research"
+authors = [
+  { name = "Danila Bredikhin" },
 ]
-dependencies = [
-    "anndata >= 0.10.8",
-    "pandas >= 1.4.0"
+requires-python = ">=3.10"
+classifiers = [
+  "Intended Audience :: Science/Research",
+  "License :: OSI Approved :: BSD License",
+  "Operating System :: OS Independent",
+  "Programming Language :: Python :: 3 :: Only",
+  "Programming Language :: Python :: 3.10",
+  "Programming Language :: Python :: 3.11",
+  "Programming Language :: Python :: 3.12",
+  "Programming Language :: Python :: 3.13",
+  "Topic :: Scientific/Engineering :: Bio-Informatics",
 ]
-dynamic = ["version"]
-
-[project.urls]
-Documentation = "https://mudata.readthedocs.io/en/latest/"
-Source = "https://github.com/scverse/mudata"
-Home-page = "https://muon.scverse.org/"
+dynamic = [ "version" ]
 
-[project.optional-dependencies]
-dev = [
-    "setuptools_scm",
+dependencies = [
+  "anndata>=0.10.8",
+  "numpy",
+  "pandas>=1.4",
+  "scipy",
+  # for debug logging (referenced from the issue template)
+  "session-info2",
 ]
-docs = [
-    "sphinx",
-    "sphinx-book-theme",
-    "pydata-sphinx-theme",
-    "readthedocs-sphinx-search",
-    "nbsphinx",
-    "sphinx_automodapi",
-    "recommonmark"
+optional-dependencies.dev = [
+  "fsspec",
+  "hatch-vcs",    # for version handling in editable mode
+  "pre-commit",
+  "twine>=4.0.2",
+  "zarr<3",
 ]
-test = [
-    "zarr<3",
-    "pytest"
+optional-dependencies.doc = [
+  "docutils>=0.8,!=0.18.*,!=0.19.*",
+  "ipykernel",
+  "ipython",
+  "myst-nb>=1.1",
+  "pandas",
+  # Until pybtex >0.24.0 releases: https://bitbucket.org/pybtex-devs/pybtex/issues/169/
+  "setuptools",
+  "sphinx>=8.1",
+  "sphinx-autodoc-typehints",
+  "sphinx-automodapi",
+  "sphinx-book-theme>=1",
+  "sphinx-copybutton",
+  "sphinx-tabs",
+  "sphinxcontrib-bibtex>=1",
+  "sphinxext-opengraph",
 ]
-
-[tool.pytest.ini_options]
-python_files = "test_*.py"
-testpaths = [
-    "./tests", # unit tests
+optional-dependencies.test = [
+  "coverage>=7.10",
+  "pytest",
+  "zarr<3",
```

Comment:
Why `zarr<3`?

Reply:
That got moved from somewhere else in this file. But good point, we should remove the bound for 0.4. But that's a separate PR/commit, I think.